plugins/state_history_plugin/tests/main.cpp

```diff
@@ -0,0 +1,2 @@
+#define BOOST_TEST_MODULE state_history_plugin
+#include <boost/test/included/unit_test.hpp>
```

Comment:
Since you're using the header-only variant now, you should remove linking to the library from here
https://github.com/AntelopeIO/leap/blob/870673840b94532ee2c07bb70d4b1a1b71182b02/plugins/state_history_plugin/tests/CMakeLists.txt#L2